src/util/psbt/mod.rs

```diff
@@ -51,7 +51,7 @@ pub type Psbt = PartiallySignedTransaction;
 
 /// A Partially Signed Transaction.
 #[derive(Debug, Clone, PartialEq, Eq)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+//#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
I know its a draft but there is too much going on in this PR for me to review it by diff, I had to check out the branch and play around with it (could just be me). The serde stuff could be done in a separate patch (even done later once you get the main bit working).

Reply:
Much of my confusion comes from incomplete understanding of the role of serde. Don't we have 2 `Serialize` Traits on PartiallySignedTransaction with this line?

Reply:
You can implement multiple traits by the same name AFAIK, but like I say somewhere else, I don't think we should implement our  `Serialize` trait anyways.